src/main/java/edu/arizona/kra/negotiations/dao/ojb/NegotiationLookupDaoOjb.java

```diff
@@ -0,0 +1,284 @@
+        fieldValues.remove(NEGOTIATION_AGE);
+
+        Map<String, String> associationDetails = getAssociationFilters( fieldValues );
+
+        //generate negotiation query criteria from user filters
+        Criteria searchCriteria = getCollectionCriteriaFromMap(new Negotiation(), fieldValues);
+
+        //add additional criteria related to associated objects - if any
+        if (!associationDetails.isEmpty()) {
+            Criteria associatedCriteria = new Criteria();
+
+            associatedCriteria.addOrCriteria( getCriteria(new Award(), NegotiationAssociationType.AWARD_ASSOCIATION, "awardNumber", associationDetails, awardTransform, Boolean.TRUE) );
+            associatedCriteria.addOrCriteria( getCriteria(new SubAward(), NegotiationAssociationType.SUB_AWARD_ASSOCIATION, "subAwardCode", associationDetails, subAwardTransform, Boolean.TRUE) );
+            associatedCriteria.addOrCriteria( getCriteria(new InstitutionalProposal(), NegotiationAssociationType.INSTITUATIONAL_PROPOSAL_ASSOCIATION, "proposalNumber", associationDetails, institutionalProposalTransform, Boolean.TRUE) );
+            associatedCriteria.addOrCriteria( getCriteria(new NegotiationUnassociatedDetail(), NegotiationAssociationType.NONE_ASSOCIATION, "negotiationUnassociatedDetailId", associationDetails, unassociatedTransform, Boolean.FALSE) );
+
+            //UofA doesn't use ProposalLogs... commenting out in case we decide to use them later on
+            //associatedCriteria.addOrCriteria( getCriteria(new ProposalLog(), NegotiationAssociationType.PROPOSAL_LOG_ASSOCIATION, "proposalNumber", associationDetails, proposalLogTransform, Boolean.FALSE) );
+
+
+            if ( !associatedCriteria.isEmpty() ) {
+                searchCriteria.addAndCriteria(associatedCriteria);
+            }
+
+        }
+        if ( StringUtils.isNotEmpty(negotiationAge) ) {
+            //add additional criteria related to negotiation age - if any
+            searchCriteria.addAndCriteria( getNegotiationAgeCriteria(negotiationAge) );
+        }
+
+        LOG.debug("Final search criteria: {}", searchCriteria);
+
+        try {
+            //add orderBy NegotiationId to show the newest on top to query
+            QueryByCriteria query = QueryFactory.newQuery(Negotiation.class, searchCriteria);
+            query.addOrderByDescending(NEGOTIATION_ID);
+
+            Long matchingResultsCount = null;
+            Integer searchResultsLimit = LookupUtils.getSearchResultsLimit(Negotiation.class);
+            if (searchResultsLimit != null) {
+                LOG.debug("getNegotiationResults searchResultsLimit={}",searchResultsLimit);
+
+                matchingResultsCount = new Long(getPersistenceBrokerTemplate().getCount(query));
+                LOG.debug("getNegotiationResults matchingResultsCount={}",matchingResultsCount);
+
+                LookupUtils.applySearchResultsLimit(Negotiation.class, searchCriteria, getDbPlatform());
+            }
```

Comment:
Adding this might be useful in the future, or even now since it's not readily apparent where the limit value is being set. 

`else { LOG.warn("SearchResultsLimit for Business Object and Application is undefined"); }`